R/simulate_data.R

```diff
@@ -0,0 +1,611 @@
+#' and `pars_t$prob_miss`.
+#'
+#' The probability of the ICE after each visit is modeled according to the following
+#' logistic regression model:
+#' `~ 1 + I(visit == 0) + ... + I(visit == n_visits-1) + I((x-alpha))` where:
+#' - `n_visits` is the number of visits (including baseline).
+#' - `alpha` is the baseline outcome mean.
+#' The term `I((x-alpha))` specifies the dependency of the probability of the ICE on
+#' the current outcome value.
+#' The corresponding regression coefficients of the logistic model are defined as follows:
+#' The intercept is set to 0, the coefficients corresponding to discontinuation after
+#' each visit for a subject with outcome equal to
+#' the mean at baseline are set according to parameters `pars_c$prob_ice1` (`pars_t$prob_ice1`),
+#' and the regression coefficient associated with the covariate `I((x-alpha))` is set
+#' to `log(pars_c$or_outcome_ice)` (`log(pars_t$or_outcome_ice)`).
+#'
+#' Please note that the baseline outcome cannot be missing nor be affected by any ICEs.
+#'
+#' @returns A `data.frame` containing the simulated data. It includes the following variables:
+#' - `id`: Factor variable that specifies the id of each subject.
+#' - `visit`: Factor variable that specifies the visit of each assessment. Visit `0` denotes
+#' the baseline visit.
+#' - `group`: Factor variable that specifies which treatment group each subject belongs to.
+#' - `outcome_bl`: Numeric variable that specifies the baseline outcome.
+#' - `outcome_noICE`: Numeric variable that specifies the longitudinal outcome assuming
+#' no ICEs.
+#' - `ind_ice1`: Binary variable that takes value `1` if the corresponding visit is
+#' affected by ICE1 and `0` otherwise.
+#' - `dropout_ice1`: Binary variable that takes value `1` if the corresponding visit is
+#' affected by the drop-out following ICE1 and `0` otherwise.
+#' - `dropout2`: Binary variable that takes value `1` if the corresponding visit is affected
+#' by the uninformative study drop-out.
+#' - `ind_ice2`: Binary variable that takes value `1` if the corresponding visit is affected
+#' by ICE2.
+#' - `outcome`: Numeric variable that specifies the longitudinal outcome including ICE1, ICE2
+#' and the intermittent missing values.
+#'
+#' @export
+simulate_data <- function(pars_c, pars_t, post_ice_traj, strategies = getStrategies()) {
+
+    assert_that(
+        has_class(pars_c, "simul_pars"),
+        has_class(pars_t, "simul_pars"),
+        msg = "`pars_c` and `pars_t` must be of class `simul_pars`"
+    )
+    validate(pars_c)
+    validate(pars_t)
+    assert_that(
+        length(pars_c$mu) == length(pars_t$mu),
+        msg = "The two groups must have the same number of visits"
+    )
+
+    # TODO: validate user-defined strategies
+    match.arg(
+        arg = post_ice_traj,
+        choices = names(strategies)
+    )
+    strategy_fun <- strategies[[which(names(strategies) == post_ice_traj)]]
```

Comment:
```suggestion
    strategy_fun <- strategies[[post_ice_traj]]
```

Isn't this a named list? Do we need to convert to indexing as opposed to just using the name ?

Reply:
Agreed